### PR TITLE
[Post Preview] Adds NewPostPreviewViewController for redesign work

### LINKED ISSUE #13180: Migrate PostPreviewViewController to WKWebView
Tasks:

- [ ] Replace UIWebView and any related code with WKWebView.
- [ ] Replace UIWebViewDelegate implementation.

### PR DESCRIPTION
**Pull request by bjtitus**
Fixes #13180 (once `postPreview` feature flag is disabled)

Part of #13197

This is intended to be a rough copy of the existing class but will change as additional functionality is added (such as Desktop/Mobile previews)

To test:

- Preview a post from Edit, Post List, or other screen
- Check that new view controller is used

PR submission checklist:

- [x] ~I have considered adding unit tests where possible.~ I think the `NewPostPreview` will undergo some changes and tests should be written once it is complete.

- [x] ~I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.~ This can be done once the feature flag is released.

### CODE REVIEW

File: WordPress/Classes/ViewRelated/Post/NewPostPreviewViewController.swift [hunk 1]
```diff
@@ -0,0 +1,252 @@
...
+
+    private func setupWebView() {
+        webView = WKWebView(frame: view.bounds)
+        webView.navigationDelegate = self
+        view.addSubview(webView)
+        view.pinSubviewToAllEdges(webView)
+    }
+
+    @objc private func dismissPreview() {
+        onClose?()
```
Comment (frosty) [inline WordPress/Classes/ViewRelated/Post/NewPostPreviewViewController.swift]: This is a slightly different behaviour to the existing view controller (not sure if that matters?). The previous one only called dismiss on the presenting view controller if there was no `onClose` block provided.

File: WordPress/Classes/ViewRelated/Post/NewPostPreviewViewController.swift [hunk 2]
```diff
@@ -0,0 +1,252 @@
...
+        view.pinSubviewToAllEdges(controller.view)
+        noResultsViewController?.didMove(toParent: self)
+    }
+}
+
+extension NewPostPreviewViewController: PostPreviewGeneratorDelegate {
+    func preview(_ generator: PostPreviewGenerator, attemptRequest request: URLRequest) {
+        startLoadAnimation()
+        var newRequest = request
+        newRequest.url = URL(string: "http://sdfsldfjlwefj.sdlfj")
```
Comment (frosty) [inline WordPress/Classes/ViewRelated/Post/NewPostPreviewViewController.swift]: Is this intentional? The existing implementation just passes the request to the webView.

File: WordPress/Classes/ViewRelated/Post/NewPostPreviewViewController.swift [hunk 3]
```diff
@@ -0,0 +1,252 @@
...
+        }
+
+        if navigationAction.request.url?.absoluteString == post.permaLink {
+            // Always allow loading the preview
+            decisionHandler(.allow)
+            return
+        }
+
+        if navigationAction.navigationType == .linkActivated {
+            UIApplication.shared.open(navigationAction.request.url!, options: [:], completionHandler: nil)
```
Comment (frosty) [inline WordPress/Classes/ViewRelated/Post/NewPostPreviewViewController.swift]: Should we unwrap the `url` here to be safe?

File: WordPress/Classes/ViewRelated/Post/NewPostPreviewViewController.swift [hunk 4]
```diff
@@ -0,0 +1,252 @@
...
+            decisionHandler(.cancel)
+            return
+        }
+
```
Comment (frosty) [inline WordPress/Classes/ViewRelated/Post/NewPostPreviewViewController.swift]: We're missing the `formSubmitted` case from the original implementation.

File: WordPress/Classes/ViewRelated/Post/NewPostPreviewViewController.swift (hunk 2 above)
Comment (bjtitus) [reply WordPress/Classes/ViewRelated/Post/NewPostPreviewViewController.swift]: Oops, accidentally pushed this when testing NoResults. 🤦‍♂

Comment (bjtitus) [reply WordPress/Classes/ViewRelated/Post/NewPostPreviewViewController.swift]: Oops, accidentally pushed this after testing NoResults 🤦‍♂

**Comment by bjtitus** [comment]
@frosty Thanks for taking a look! I did a little extra clean up and fixed those issues you had mentioned.

Should be ready for review now.

**Comment by bjtitus** [comment]
@frosty I went ahead and changed all of the presentations to modals and now always show the done button.

The next set of changes will use WebKitViewController anyway so that class will end handling all of the `navigationItem` set up.

File: WordPress/Classes/ViewRelated/Post/NewPostPreviewViewController.swift [hunk 5]
```diff
@@ -47,6 +44,7 @@ class NewPostPreviewViewController: UIViewController {
     }()
 
     private lazy var doneBarButtonItem: UIBarButtonItem = {
+//        let buttonItem = UIBarButtonItem(barButtonSystemItem: UIBarButtonItem.SystemItem.done, target: self, action: #selector(dismissPreview))
```
Comment (frosty) [inline WordPress/Classes/ViewRelated/Post/NewPostPreviewViewController.swift]: Can we just remove this line?

**Pull request merged**